hikari/internal/deprecation.py

```diff
@@ -66,26 +69,37 @@ def warn_deprecated(obj: typing.Any, additional_information: str, /, *, stack_le
     )
 
 
-def deprecated(version: str, additional_information: str) -> typing.Callable[[T], T]:
+def deprecated(deprecated_version: str, removal_version: str, additional_information: str) -> typing.Callable[[T], T]:
     """Mark a function or object as being deprecated.
 
     Parameters
     ----------
-    version: typing.Any
+    deprecated_version: str
         The version this function or object is deprecated in.
+    removal_version: str
+        The version this function or object will be removed in.
     additional_information: str
         Additional information on the deprecation for the user.
     """
 
     def decorator(obj: T) -> T:
+        # No, this will not cause issues.
+        # Thanks to CI and our deploy pipeline, we will be able to catch these and remove them before deployment
+        if ux.HikariVersion(hikari_about.__version__) >= ux.HikariVersion(removal_version):
```

Comment:
When we follow a proper release schedule with proper versioning, this shouldnt be a problem, as the versions they are removed in will be mayor ones.

For now, where we only do dev releases, it will cause the CI to fail, but then we can quickly fix it.

-----

We really might want to have proper release schedule and figure out whether we want to have supported versions that get important fixes, etc.

(Ik you have been talking about this for a long time now, its just I dont have any concrete idea on how to do it properly. I am fully open to ideas an opinions)

Reply:
> For now, where we only do dev releases, it will cause the CI to fail, but then we can quickly fix it.

This would still be kind of annoying though, since it essentially means we both have to be active for a version to be released so we can deal with reviewing and merging any necessary follow up prs

Reply:
I don't really have any good systems for handling strict version based deadlines for stuff like this tbh; for my projects I'd usually give stuff a time based deprecation (so say 6 months) while they're in an alpha/beta state or if they're using a time based versioning scheme and once its stable I wouldn't really need a system like this which blares at me that a deprecated thing is still around since during a major version bump I'm going to have to comb through to find stuff to remove anyways since you can't have a system like this for arguments and it might actually give a false sense of security (possibly leading to stuff being missed) now that I think about it.

Reply:
I am open to changing it to anything. I just cant think of a nice way that will warn us whenever we need to remove it.

I have thought about leaving the tests that will test the deprecation and then have `warn_deprecated` raise the error if needed, but that seemed a bit too much as it would make all deprecated calls slower.

Reply:
>  would make all deprecated calls slower.

Don't really see why that'd matter

> I am open to changing it to anything. I just cant think of a nice way that will warn us whenever we need to remove it.

The current system you've got in this PR doesn't achieve this since it only works for deprecated methods and not deprecated parameters

Reply:
> The current system you've got in this PR doesn't achieve this since it only works for deprecated methods and not deprecated parameters

Fixed that :+1:

Reply:
tw how'd you fix that?